Makefile

```diff
@@ -2,34 +2,58 @@ TARGETS = naive cuda cupla kokkos oneapi
 
 .PHONY: all debug clean $(TARGETS)
 
+# general rules and targets
+all: $(TARGETS)
+
+debug: $(TARGETS:%=%-debug)
+
+clean:
+	rm -f main-* debug-*
+
+# configure external tool here
+BOOST_BASE  :=
+TBB_BASE    :=
+CUDA_BASE   := /usr/local/cuda
+ALPAKA_BASE := /usr/local/alpaka/alpaka
+CUPLA_BASE  := /usr/local/alpaka/cupla
+
+# host compiler
 CXX := g++
-CXX_FLAGS := -O2 -std=c++14 -ftemplate-depth-512
+CXX_FLAGS := -O2 -std=c++14
 CXX_DEBUG := -g
 
-CUDA_BASE := /usr/local/cuda
+# CUDA compiler
 NVCC := $(CUDA_BASE)/bin/nvcc -ccbin $(CXX)
 NVCC_FLAGS := -O2 -std=c++14 --expt-relaxed-constexpr -w --generate-code arch=compute_35,code=sm_35
 NVCC_DEBUG := -g -lineinfo
 
-ALPAKA_BASE  := /usr/local/alpaka/alpaka
-ALPAKA_FLAGS := -DALPAKA_DEBUG=0 -I$(ALPAKA_BASE)/include
-CUPLA_BASE   := /usr/local/alpaka/cupla
-CUPLA_FLAGS  := $(ALPAKA_FLAGS) -I$(CUPLA_BASE)/include
-
-ONEAPI_CXX := $(shell which dpcpp 2> /dev/null)
+# boost flags
+ifdef BOOST_BASE
+BOOST_CXX_FLAGS := -I$(BOOST_BASE)/include
+else
+BOOST_CXX_FLAGS :=
+endif
 
-GREEN := '\033[32m'
-RED := '\033[31m'
-RESET := '\033[0m'
+# TBB flags
+ifdef TBB_BASE
+TBB_CXX_FLAGS := -I$(TBB_BASE)/include
+TBB_LD_FLAGS  := -L$(TBB_BASE)/lib -ltbb -lrt
+else
+TBB_CXX_FLAGS :=
+TBB_LD_FLAGS  := -ltbb -lrt
```

Comment:
Is the assumption that if `TBB_BASE` is not defined, TBB is in a "standard location" and thus `-I$(TBB_BASE)/include` and `-L$(TBB_BASE)/lib` do not need to be specified?